### DISCUSSION TITLE
v2.7.2

### DISCUSSION THREAD

**Discussion by Limraj** [SCADA-LTS/Scada-LTS#2371] [General]
## 

The changelist is listed below:

### 📄️ GitHub issues/pullrequest list overview:
- #731 Error while enabling or disabling data source via script
- #966 Add missing AMQP datasource
- #1694 Modbus Serial is not working 
- #1894 Views caching
- #1734 Vulnerabilities from ScadaBR
- #2111 Fixed using setTimeout in common.js (#2081)
- #2216 Improvement of error presentation on cmp component in version 3
- #2218 MQTT Protocol support / subscribe to a MQTT Broker(s)
- #2232 Validate state when defining Multistate Graphic Component
- #2257 Independence from package com.serotonin.web.email from seroUtils
- #2272 Corrected edit Graphical View for admin user
- #2275 Separation editing and refreshing Graphical View opened in multiple tabs 
- #2284 UserComments caching
- #2304 Created api to retrieve detailed information about active threads
- #2320 Added warnning messages for Data Points in Graphical View
- #2331 Fixed npm process for windows7 and gradle
- #2334 Fix code scanning alert - No clone method - AmqpDataSourceVO
- #2335 Fix code scanning alert - No clone method - MqttDataSourceVO
- #2336 Code fixed according to the CodeQL report
- #2343 Corrected msg logging error for Data Source Messaging
- #2347 Fixed websocket send message for event with handlers
- #2349 Fixed Authentication REST API

ScadaBR.war file checksum: `bc1b6888519d147a61c3470a38866121 `

<hr /><em>This discussion was created from the release <a href='https://github.com/SCADA-LTS/Scada-LTS/releases/tag/v2.7.2'>v2.7.2</a>.</em>

**Comment by ardy2005** [comment]
Hi, I installed version 2.6.17 on kylin Ubuntu20.04 and it works fine.But when I change the ScadaBR.war file in "/var/lib/tomcat9 / webapps/" directory for new version 2.7.2 file, use the “admin” login system, note the information:"**New page is available**".
![new page available](https://user-images.githubusercontent.com/45162770/201903285-074f338c-3db2-4b6e-b147-7d1d49056f4c.jpg)

To enter "graphic views" page,the prompt is as follows: A server error has occured.
![server error](https://user-images.githubusercontent.com/45162770/201903489-242c2953-25ef-4fd9-b7c5-a178d50ab23c.jpg)

and the Components list is empty.
![empty](https://user-images.githubusercontent.com/45162770/201903555-b1fd19ff-9356-43dd-8fb8-6a6e054f51a7.jpg)

MySql version :5.7.34
Tomcat9
web Browser is Firefox

**Comment by Limraj** [comment]
Hi @ardy2005 ,
there was a problem when the admin edited a view he didn't own, i.e. the view was created by another user, and the profile was saved in the admin, even if it had an admin set, but this bug has been fixed in this version:
https://github.com/SCADA-LTS/Scada-LTS/pull/2273/files

1. Probably you just need to do a hard reset browser, the problem should not reoccur, then you update the cached javascript causing this error. How: https://fabricdigital.co.nz/blog/how-to-hard-refresh-your-browser-and-clear-cache

2. Also make sure you have one version in webapps, i.e. ideally, stop tomcat, delete everything in that folder, then paste the war with the new version, you should have one file in there at the moment. Then start tomcat.

Link "New page is available" - It allows you to switch to the new interface we are creating. You can also access it via the "Modern interface" button. It doesn't have all the features of the old interface yet, it's under development.

Be sure to let us know if these steps helped resolve the issue. Regards.

**Reply by ardy2005** [reply]
Thanks, I see, "New Page is available" is just a hint, it's not a problem.
For the problem that prompts "A server error has occured", I tried the following steps:
1. sudo systemctl stop tomcat9
2. Delete the scadalts database from mysql
3. Re-create the scadalts database.
4. Delete the ScadaBR.war file and ScadaBR directory in the /var/lib/tomcat9/webapps directory
5. Copy the new ScadaBR.war file to this directory
6. sudo systemctl start tomcat9
But the error is still displayed

I later reinstalled the operating system, followed the documentation steps to configure the software, and copied the 2.7.2.war file so that it worked without error. I'm going to do it this way for now.

**Reply by Limraj** [reply]
1. Just upload zip file with your configuration (export the project), I'll see at my place.
2. In addition, you can also try on a second browser where you did not open Scada to rule out a problem with cached js.
3. You can also check catalina.out / mango.log for any exceptions. (/var/lib/tomcat9/logs)
4. Check the console in the browser (right click on the page, click 'Inspect' -> 'Console') send the screen.
5. You can also send the md5 hash of your war. (command md5sum in ubuntu)

Unfortunately, with the current state of knowledge of the problem, I cannot reproduce it, so I need more information.
If you follow these steps, you provide data, there's a good chance we'll figure out what's going on.

**Comment by Limraj** [comment]
If that doesn't help, it still shows up, then 
1. Export the project:
Tab "Export/Import" -> Section "Export Project (Download)" -> Click button "Export project"
Attach the generated zip file here.
2. Screenshot from console browser: Click right mouse -> "Inspect" -> "Console"
